src/global/tracking/TrackPropagation_factory.cc

```diff
@@ -0,0 +1,194 @@
+#include <services/geometry/acts/ACTSGeo_service.h>
+
+#include <Acts/EventData/MultiTrajectoryHelpers.hpp>
+#include <Acts/Surfaces/RadialBounds.hpp>
+
+#include <extensions/spdlog/SpdlogExtensions.h>
+#include <services/log/Log_service.h>
+#include <spdlog/fmt/ostr.h>
+
+#include <edm4eic/TrackPoint.h>
+#include <edm4eic/TrackSegment.h>
+
+void eicrecon::TrackPropagation_factory::Init() {
+
+    auto app = GetApplication();
+
+    // SpdlogMixin logger initialization, sets m_log
+    InitLogger(app, GetTag());
+
+    auto acts_service = GetApplication()->GetService<ACTSGeo_service>();
+    m_track_propagation_algo.init(acts_service->actsGeoProvider(), logger());
+
+    m_geoSvc = app->template GetService<JDD4hep_service>();
+
+    SetPropagationSurfaces();
+
+}
+
+void eicrecon::TrackPropagation_factory::Process(const std::shared_ptr<const JEvent> &event) {
+
+    auto trajectories = event->Get<eicrecon::TrackingResultTrajectory>(GetInputTags()[0]);
+
+    edm4eic::TrackSegmentCollection propagated_tracks;
+
+    for(auto traj: trajectories) {
+        edm4eic::MutableTrackSegment this_propagated_track;
+        for(size_t isurf = 0; auto surf: m_target_surface_list) {
+            auto prop_point = m_track_propagation_algo.propagate(traj, surf);
+            if(!prop_point) continue;
+            prop_point->detectorID = m_target_detector_ID[isurf];
+            prop_point->surfaceID = m_target_surface_ID[isurf];
+            this_propagated_track.addToPoints(*prop_point);
+            isurf++;
+        }
+        propagated_tracks.push_back(this_propagated_track);
+    }
+
+    SetCollection<edm4eic::TrackSegment>(GetOutputTags()[0], std::move(propagated_tracks));
+
+}
+
+
+void eicrecon::TrackPropagation_factory::SetPropagationSurfaces() {
+
+    auto transform = Acts::Transform3::Identity();
+
+    // shift projection surface to average cluster depth
+    // shift in z for endcaps, shift in r for barrel
+    double ECAL_avgClusterDepth = 50.;	  // mm
+    double HCAL_avgClusterDepth = 150.;   // mm
+
+    // extend surfaces by ten percent for tracks close to the edge
+    // extend in r for endcaps, extend in z for barrel
+    double extend = 1.1;
+
+    // Create propagation surface for BEMC
+    std::shared_ptr<Acts::CylinderSurface> m_BEMC_prop_surface1;
+    std::shared_ptr<Acts::CylinderSurface> m_BEMC_prop_surface2;
+    const double BEMC_R     = m_geoSvc->detector()->constant<double>("EcalBarrel_rmin") / dd4hep::mm;
+    const double BEMC_halfz = (std::max(m_geoSvc->detector()->constant<double>("EcalBarrelBackward_zmax"), m_geoSvc->detector()->constant<double>("EcalBarrelForward_zmax")) / dd4hep::mm) * extend;
+    auto BEMC_Trf         = transform * Acts::Translation3(Acts::Vector3(0, 0, 0));
+    m_BEMC_prop_surface1  = Acts::Surface::makeShared<Acts::CylinderSurface>(BEMC_Trf, BEMC_R, BEMC_halfz);
+    m_BEMC_prop_surface2  = Acts::Surface::makeShared<Acts::CylinderSurface>(BEMC_Trf, BEMC_R + ECAL_avgClusterDepth, BEMC_halfz);
```

Comment:
Type is evident from the RHS
```suggestion
    auto m_BEMC_prop_surface1  = Acts::Surface::makeShared<Acts::CylinderSurface>(BEMC_Trf, BEMC_R, BEMC_halfz);
    auto m_BEMC_prop_surface2  = Acts::Surface::makeShared<Acts::CylinderSurface>(BEMC_Trf, BEMC_R + ECAL_avgClusterDepth, BEMC_halfz);
```